libs/agno/agno/utils/media.py

```diff
@@ -185,3 +185,139 @@ def download_knowledge_filters_sample_data(
         )
         file_paths.append(str(download_path))
     return file_paths
+
+
+def reconstruct_image_from_dict(img_data):
+    """
+    Reconstruct an Image object from dictionary data.
+
+    Handles both base64-encoded content (from database) and regular image data (url/filepath).
+    """
+    from agno.media import Image
+
+    if isinstance(img_data, dict):
+        # If content is base64 string, decode it back to bytes
+        if "content" in img_data and isinstance(img_data["content"], str):
+            return Image.from_base64(
+                img_data["content"],
+                id=img_data.get("id"),
+                mime_type=img_data.get("mime_type"),
+                format=img_data.get("format"),
+                detail=img_data.get("detail"),
+                original_prompt=img_data.get("original_prompt"),
+                revised_prompt=img_data.get("revised_prompt"),
+                alt_text=img_data.get("alt_text"),
+            )
+        else:
+            # Regular image (filepath/url)
+            return Image(**img_data)
+    return img_data
```

Comment:
We may want some error handling here. If we have a list of images and reconstructing one fails we probably should keep going.